test/common/filesystem/filesystem_impl_test.cc

```diff
@@ -12,6 +12,10 @@
 namespace Envoy {
 namespace Filesystem {
 
+static constexpr FlagSet DefaultFlags{
+    1 << Filesystem::File::Operation::Read | 1 << Filesystem::File::Operation::Write |
+    1 << Filesystem::File::Operation::Create | 1 << Filesystem::File::Operation::Append};
```

Comment:
again I do not understand why we are specifying all these seemingly contradictory modes.

Reply:
The same as the logic before this change. Before this change, the open method will declare all modes and use as default flags.  see here:

https://github.com/envoyproxy/envoy/blob/a9493281bd91b797b0566fea1a8b22d61d11af90/source/common/filesystem/posix/filesystem_impl.cc#L30-L36